packages/@react-spectrum/listbox/src/ListBoxOption.tsx

```diff
@@ -74,20 +74,28 @@ export function ListBoxOption<T>(props: OptionProps<T>) {
     ? <Text>{rendered}</Text>
     : rendered;
 
+  let isKeyboardModality = isFocusVisible();
+
   return (
     <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
       <div
-        {...mergeProps(optionProps, shouldFocusOnHover ? {} : hoverProps)}
+        {...mergeProps(optionProps, shouldFocusOnHover ? hoverProps : {})}
         ref={ref}
+        // aria 1.2 requires the focused combobox item to be aria-selected=true in order for Voiceover to announce it.
+        // Still has issues with combobox with sections, will need to file an issue with apple
+        aria-selected={(shouldUseVirtualFocus && isFocused) || optionProps['aria-selected']}
```

Comment:
Are we sure about this? Selection isn't the same as focus. We also have aria-selected set already based on the actual selection. Plus, I see that VoiceOver still doesn't announce the focused item sometimes...

Reply:
Yeah, as per https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html in the "Listbox Popup" section, the focused item should have `aria-selected` (you can see this in their example as well when you navigate through the list via the arrow keys). Removing it here from the ListBoxOption causes it to no longer announce in Voiceover for me when I do the same arrow key navigation. 

I didn't account for the aria-selected that is set for the actual selected item though, thats a good catch, not sure what to do about that. Any repro steps for the not announcing focused item issue?